app/cli.h

```diff
@@ -0,0 +1,206 @@
+#ifndef BERGAMOT_APP_CLI_H
+#define BERGAMOT_APP_CLI_H
+#include <cstdlib>
+#include <future>
+#include <iostream>
+#include <sstream>
+
+#include "common/definitions.h"
+#include "common/timer.h"
+#include "common/utils.h"
+#include "marian.h"
+#include "translator/byte_array_util.h"
+#include "translator/parser.h"
+#include "translator/response.h"
+#include "translator/response_options.h"
+#include "translator/service.h"
+
+namespace marian {
+namespace bergamot {
+
+// marian::bergamot:: makes life easier, won't need to prefix it everywhere and these classes plenty use constructs.
+
+namespace app {
+
+/// Interface for command-line applications. All applications are expected to use the Options based parsing until
+/// someone builds a suitable non-marian based yet complete replacement.
+
+class CLIAppInterface {
+ public:
+  virtual void execute(Ptr<Options> options) = 0;  // Make pure virtual, attempt to compile at MSVC
+};
+
+/// Previously bergamot-translator-app. Provides a command-line app on native which executes the code-path used by Web
+/// Assembly. Expected to be maintained consistent with how the browser (Mozilla through WebAssembly) dictates its API
+/// and tests be intact. Also used in [bergamot-evaluation](https://github.com/mozilla/bergamot-evaluation).
+/// @param [cmdline]: Options to translate passed down to marian through Options.
+/// @param [stdin] sentences as lines of text.
+/// @param [stdout] translations for the sentences supplied in corresponding lines
+class WASM : public CLIAppInterface {
+ public:
+  void execute(Ptr<Options> options) {
+    std::string config = options->asYamlString();
+
+    // Route the config string to construct marian model through TranslationModel
+    Service model(config);
```

Comment:
https://github.com/browsermt/bergamot-translator/blob/8bec1b7b6ba30c522c102621e55b8cbe4bd051da/src/translator/service.h#L70

Service takes options as well, why do we need to convert back and forth?

Reply:
Objective is to test mozilla's path. 
cmdline -> Options -> std::string -> invoke std::string constructor.

Reply:
Oh god... Can we, like, not?!! At the very least comment the shit out of it.

Reply:
Yes, I understand the horror of converting structured data back into string and then reconverting it either, but you gotta test that string based constructor somewhere. :'D

You see what I have to deal with on a daily basis?  I'll go with comment the hell out option, thanks.

Reply:
> Is there really no other usage of this codepath?

Nope. I like `Ptr<Options>` and will go with it any day in my dev. Would also not mind a struct equivalent that's not marian-dependent, but I don't see it a justified investment of time to write a new one. Only Mozilla uses it. I also use it in the cmdapp which tests Mozilla's codepath, coz, obvious reasons.